Formula/r/recc.rb

```diff
@@ -0,0 +1,181 @@
+class Recc < Formula
+  desc "Remote Execution Caching Compiler"
+  homepage "https://buildgrid.gitlab.io/recc"
+  url "https://gitlab.com/BuildGrid/buildbox/buildbox/-/archive/1.2.20/buildbox-1.2.20.tar.gz"
+  sha256 "5ad5ccbad1250accc984823bd5d78604441c7711320b0569889f45f329ed6311"
+  license "Apache-2.0"
+  head "https://gitlab.com/BuildGrid/buildbox/buildbox.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "tomlplusplus" => :build
+  depends_on "abseil"
+  depends_on "c-ares"
+  depends_on "glog"
+  depends_on "grpc"
+  depends_on "openssl@3"
+  depends_on "protobuf"
+  depends_on "re2"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "gflags"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "util-linux"
+  end
+
+  def install
+    buildbox_cmake_args = %W[
+      -DCASD=ON
+      -DCASD_BUILD_BENCHMARK=OFF
+      -DCASDOWNLOAD=OFF
+      -DCASUPLOAD=OFF
+      -DFUSE=OFF
+      -DLOGSTREAMRECEIVER=OFF
+      -DLOGSTREAMTAIL=OFF
+      -DOUTPUTSTREAMER=OFF
+      -DRECC=ON
+      -DREXPLORER=OFF
+      -DRUMBA=OFF
+      -DRUN_BUBBLEWRAP=OFF
+      -DRUN_HOSTTOOLS=ON
+      -DRUN_OCI=OFF
+      -DRUN_USERCHROOT=OFF
+      -DTREXE=OFF
+      -DWORKER=OFF
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    %w[cc c++ gcc g++ clang clang++].each do |compiler|
+      (bin/"recc-#{compiler}").write <<~EOS
+        #!/usr/bin/env sh
+
+        #{bin}/recc $(command -v #{compiler}) "$@"
+      EOS
+    end
+
+    # Generate recc-server start helper script
+    (bin/"recc-server").write <<~EOS
+      #!/usr/bin/env sh
+
+      # Get current soft limit for open files
+      current_limit=$(ulimit -Sn)
+
+      REQUIRED_FILES_LIMIT=384
+      # Check if the current limit is less than REQUIRED_FILES_LIMIT
+      if [ "$current_limit" -lt ${REQUIRED_FILES_LIMIT} ]; then
+        ulimit -Sn ${REQUIRED_FILES_LIMIT}
```

Comment:
It doesn't seem right for us to write a script that mutates the user's state when run.

If this is required, then users should know to do this from reading upstream documentation. It _might_ be okay to set it in the `service` block by setting the `SoftResourceLimits` property of the generated plist. (Though this may not yet be implemented in `brew`.)

Reply:
The updated ulimit only applies to the `recc-server`. I confirmed it by running the following

```bash
ulimit -Sn; ulimit -Sn 256; ulimit -Sn;recc-server &; pid=$! ; ulimit -Sn; kill $pid;ulimit -Sn
```